### ISSUE TITLE
`unocss/postcss` does not recognise the contents of the `packages/**` in next monorepo

### ISSUE THREAD

**Issue by blacksand** [unocss/unocss#4094]
### UnoCSS version

0.62.0

### Describe the bug

The monorepo structure is as follows, worked as expected in `0.61.9`

After upgrade to `0.62.0`, the contents of the `packages/**` cannot be extracted.

```sh
.
├── apps
│   └── web   # <-- next.js app
│       ├── src
│       ├── postcss.config.mjs
│       └── uno.config.mjs
└── packages
    ├── ...
    ├── ui-echarts
    └── ui-web
```
```js
// postcss.config.mjs
const config = {
  plugins: [
    '@unocss/postcss',
  ],
}

export default config
```

```js
// uno.config.mjs
import path from 'node:path'
import {fileURLToPath} from 'node:url'
import {defineConfig, presetUno, transformerDirectives} from 'unocss'

const projectRoot = path.dirname(fileURLToPath(import.meta.url))
const workspaceRoot = path.resolve(projectRoot, '../..')

// console.log({projectRoot, workspaceRoot})

const config = defineConfig({
  presets: [
    presetUno(),
  ],
  transformers: [
    transformerDirectives(),
  ],
  content: {
    filesystem: [
      `${projectRoot}/src/**/*.{tsx,html}`,
      `${workspaceRoot}/packages/ui-*/src/**/*.{tsx,html}`
    ]
  }
})

export default config
```

The image above is the output of v0.61.9 ✅ and the following is v0.62.0 ❌
![CleanShot 2024-08-22 at 19 51 46@2x](https://github.com/user-attachments/assets/46531b27-c02d-4860-908e-a531f989960c)

### Reproduction

[blacksand/unocss-postcss-issue-report](https://github.com/blacksand/unocss-postcss-issue-report?tab=readme-ov-file)

### System Info

Node: 20.16.0
OS: darwin-x64
pnpm: 9.7.1

### Validations

- [x] Read the [Contributing Guidelines](https://github.com/unocss/unocss/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't [already an issue](https://github.com/unocss/unocss/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/unocss/unocss/discussions) or join our [Discord Chat Server](https://chat.antfu.me).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by SuperchupuDev** [comment]
it should work if you don't use absolute paths i think

**Comment by blacksand** [comment]
> it should work if you don't use absolute paths i think

I've tried but with the same result.

```js
const config = defineConfig({
  ....
  content: {
    filesystem: [
      'src/**/*.{tsx,html}',
      '../../packages/ui-*/src/**/*.{tsx,html}'
    ]
  }
})
```

**Comment by SuperchupuDev** [comment]
oh, that's an edge case that made me open SuperchupuDev/tinyglobby#18, but i hadn't seen any real user encountering it. guess i'll merge it and release a fix for the globbing library used here

**Comment by antfu** [comment]
Thanks for the prompt fix @SuperchupuDev! Let me know when you made the release so we can update and close this issue.

**Comment by blacksand** [comment]
Thanks @SuperchupuDev, this is a common requirement for monorepo projects, and it would be great to have support for absolute paths!

**Comment by SuperchupuDev** [comment]
@blacksand can you add the following field to your `package.json`, reinstall dependencies, and try again with the relative path approach? to see if publishing a new release would fix it

```json
  "resolutions": {
    "tinyglobby": "https://pkg.pr.new/tinyglobby@20"
  },
```

> this is a common requirement for monorepo projects, and it would be great to have support for absolute paths!

the library does support absolute patterns, but in a rather limited way (cannot match anything outside the cwd), so any absolute pattern that doesn't start with the cwd will return an empty match. the logic added to support `../` should be able to make improving this easier though, so it's definitely not impossible to add 👍

**Comment by blacksand** [comment]
yes, it's working fine now, thanks

**Comment by SuperchupuDev** [comment]
great, will publish the new release then

**Comment by SuperchupuDev** [comment]
released as `0.2.3`

**Comment by SuperchupuDev** [comment]
good news @blacksand! i've just released full support for absolute paths in the globbing library used here, which means your original code described here should work now too once you update your dependencies 🎉
